### PR TITLE
Support key fields validation

### PR DESCRIPTION
**Pull request by scottming**
This pr is part of the section on key_feilds validation for https://github.com/DivvyPayHQ/absinthe_federation/issues/19, and I was going to go ahead and implement the section on external. But on closer inspection of the main branch of Apollo Federation. I see that they will be removing this type of validation in the future, so I'm not sure if I want to continue.

The error implemented code is [here](https://github.com/apollographql/federation/blob/main/internals-js/src/error.ts)

[erro2.md](https://github.com/DivvyPayHQ/absinthe_federation/files/7832538/erro2.md)

generate command is: `npm run error-code-doc`

If I need to continue with some validation, it would be more appropriate to target both federation1 and 2. Do you have any suggestions on this? @kdawgwilk

### CODE REVIEW

File: lib/absinthe/federation/schema/phase/key_fields_must_be_exist.ex [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    with {:ok, tokens} <- Absinthe.Lexer.tokenize("{ " <> nested_key <> " }"),
+         {:ok, parsed} <- :absinthe_parser.parse(tokens) do
+      access = [Access.key(:definitions), Access.at(0), Access.key(:selection_set), Access.key(:selections)]
+      get_in(parsed, access)
+    end
+  end
+
+  def explanation(key, object) do
+    """
+    The @key #{inspect(key)} is not exist in #{inspect(object.identifier)} object.
```
Comment (kdawgwilk) [inline lib/absinthe/federation/schema/phase/key_fields_must_be_exist.ex:113]:
```suggestion
    The @key #{inspect(key)} does not exist in #{inspect(object.identifier)} object.
```

File: lib/absinthe/federation/schema/phase/key_fields_must_be_exist.ex [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    """
+  end
+
+  def explanation(field, _object, key_fields) do
+    """
+    The field #{inspect(field)} of @key #{inspect(key_fields)} is not exist.
```
Comment (kdawgwilk) [inline lib/absinthe/federation/schema/phase/key_fields_must_be_exist.ex:119]:
```suggestion
    The field #{inspect(field)} of @key #{inspect(key_fields)} does not exist.
```

**Review by kdawgwilk — CHANGES_REQUESTED** [review]

**Comment by kdawgwilk** [comment]
federation v2 is still 3-6 months out so we can focus purely on v1 for now

**Comment by scottming** [comment]
> federation v2 is still 3-6 months out so we can focus purely on v1 for now

Ok, I'll submit another pr for the external thing.

**Review by kdawgwilk — APPROVED** [review]

**Review by MrGrinst — APPROVED** [review]

**Pull request merged**